hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -335,7 +335,22 @@ int MDMParser::sendFormattedWithArgs(const char* format, va_list args) {
 bool MDMParser::_atOk(void)
 {
     sendFormated("AT\r\n");
-    return (RESP_OK == waitFinalResp(nullptr, nullptr, AT_TIMEOUT));
+    int resp = waitFinalResp(nullptr, nullptr, AT_TIMEOUT);
+    bool atok = (resp == RESP_OK);
+
+    // R410 Power Savings Mode currently disabled, therefore this only affects all other modems
+    if (!atok && resp == WAIT && _dev.dev != DEV_SARA_R410 && _dev.lpm == LPM_ACTIVE) {
```

Comment:
Four conditions is intimidating. This compound `if` could benefit from removing the `!atok` check. It is redundant, because it is always true when `resp == WAIT`.

Reply:
I really love the simplified logic!

However, I still have an issue with checking the _dev.dev != DEV_SARA_R410. It looks like this logic would be covered, and more importantly ABSTRACTED, by _dev.lpm == LPM_ACTIVE.